front/pages/api/w/[wId]/vaults/[vId]/data_sources/index.ts

```diff
@@ -0,0 +1,542 @@
+import type { DataSourceViewType, WithAPIErrorResponse } from "@dust-tt/types";
+import {
+  CONNECTOR_PROVIDERS,
+  ConnectorConfigurationTypeSchema,
+  ConnectorsAPI,
+  CoreAPI,
+  DEFAULT_QDRANT_CLUSTER,
+  dustManagedCredentials,
+  EMBEDDING_CONFIGS,
+  ioTsParsePayload,
+  isConnectorProvider,
+  isDataSourceNameValid,
+  sendUserOperationMessage,
+  WebCrawlerConfigurationTypeSchema,
+} from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import config from "@app/lib/api/config";
+import { getDataSource, getDataSources } from "@app/lib/api/data_sources";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { getOrCreateSystemApiKey } from "@app/lib/auth";
+import {
+  getDefaultDataSourceDescription,
+  getDefaultDataSourceName,
+  isConnectorProviderAllowedForPlan,
+  isConnectorProviderAssistantDefaultSelected,
+  isValidConnectorSuffix,
+} from "@app/lib/connector_providers";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
+import { ServerSideTracking } from "@app/lib/tracking/server";
+import { isDisposableEmailDomain } from "@app/lib/utils/disposable_email_domains";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+
+// Sorcery: Create a union type with at least two elements to satisfy t.union
+const ConnectorProviderCodec = t.union([
+  t.literal(CONNECTOR_PROVIDERS[0]),
+  t.literal(CONNECTOR_PROVIDERS[1]),
+  ...CONNECTOR_PROVIDERS.slice(2).map((provider) => t.literal(provider)),
+]);
+
+const PostManagedDataSourceRequestBodySchema = t.type({
+  provider: ConnectorProviderCodec,
+  connectionId: t.union([t.string, t.undefined]),
+  configuration: ConnectorConfigurationTypeSchema,
+});
+
+const PostStaticDataSourceRequestBodySchema = t.type({
+  name: t.string,
+  description: t.string,
+  assistantDefaultSelected: t.boolean,
+});
+
+type PostVaultDataSourceResponseBody = {
+  dataSourceView: DataSourceViewType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<PostVaultDataSourceResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  const user = auth.user();
+
+  if (!owner || !plan || !user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const vault = await VaultResource.fetchById(auth, req.query.vId as string);
```

Comment:
We should not cast to string, this is not accurate.
Instead we should do something like this:
```suggestion
  const {vId} = query;
  if (typeof vId !== "string") {
   // Return error.
  }
  
  const vault = await VaultResource.fetchById(auth, vId);
```

Reply:
I'm not sure why we should not do this TBH.  
Anyway since we do that everywhere including in the other vault routes, I'll keep as is, I'd rather be consistent.
Let's refactor that everywhere in a dedicated PR if we should stop casting query param as is in routes 🙏🏻